### PR TITLE
🐛 Table overflow problem

### PR DESCRIPTION
**Pull request by hlminh2000**
**Description of changes**

Fixes the table component overflow problem.
- **The problem:**
  - `react-table` expects to receive an explicit width to handle horizontal-scroll properly
- **The solution:**
  - make the `uikit/Table` component expect a `parentRef` prop to sync up `react-table`'s width with the parent
  - implements this new requirement on every table in the app (identified by TS errors)

**Type of Change**

- [x] Bug
- [ ] Styling
- [ ] New Feature

**Checklist before requesting review:**

- [ ] Matches design:

  - component sizes, spacing, and styles
  - font size, weight, colour
  - spelling has been double checked

- [ ] New uikit components have Storybook stories
- [ ] Feature is minimally responsive.
- [x] Manual testing of UI feature.
- [ ] Selenium test is completed and passing.

### CODE REVIEW

File: components/pages/submission-system/program-clinical-submission/FilesNavigator/FileRecordTable.tsx [hunk 1]
```diff
@@ -166,8 +167,11 @@ export default ({
 
   return (
     <div
+      ref={containerRef}
+      className="yo"
```
Comment (ciaranschutte) [inline components/pages/submission-system/program-clinical-submission/FilesNavigator/FileRecordTable.tsx]: @hlminh2000 what up dawg?

Comment (hlminh2000) [reply components/pages/submission-system/program-clinical-submission/FilesNavigator/FileRecordTable.tsx]: lol nice catch :P

File: uikit/Table/index.tsx [hunk 2]
```diff
@@ -126,17 +131,26 @@ function Table<Data = { [k: string]: any }>({
 
   return (
     <StyledTable
-      css={css`
-        width: ${width}px;
-      `}
+      style={{
+        // this is written with style object because css prop someone only applies to the header
```
Comment (ciaranschutte) [inline uikit/Table/index.tsx]: css prop someone only? @hlminh2000

Comment (hlminh2000) [reply uikit/Table/index.tsx]: hmmm yeah someone... not sure who :P

**Pull request merged**